content/docs/context.md

```diff
@@ -4,220 +4,92 @@ title: Context
-};
+```js
+const {Provider, Consumer} = React.createContext(defaultValue);
 ```
 
-By adding `childContextTypes` and `getChildContext` to `MessageList` (the context provider), React passes the information down automatically and any component in the subtree (in this case, `Button`) can access it by defining `contextTypes`.
+Creates a `{ Provider, Consumer }` pair.
 
-If `contextTypes` is not defined, then `context` will be an empty object.
+Optionally accepts a default value to be passed to Consumers without a Provider ancestor.
 
-## Parent-Child Coupling
+### `Provider`
 
-Context can also let you build an API where parents and children communicate. For example, one library that works this way is [React Router V4](https://reacttraining.com/react-router):
+```js
+<Provider value={/* some value */}>
+```
 
-```javascript
-import { BrowserRouter as Router, Route, Link } from 'react-router-dom';
+A React component that allows Consumers to subscribe to context changes.
 
-const BasicExample = () => (
-  <Router>
-    <div>
-      <ul>
-        <li><Link to="/">Home</Link></li>
-        <li><Link to="/about">About</Link></li>
-        <li><Link to="/topics">Topics</Link></li>
-      </ul>
+Accepts a `value` prop to be passed to Consumers that are descendants of this Provider. One Provider can be connected to many Consumers. Providers can be nested to override values deeper within the tree.
 
-      <hr />
+### `Consumer`
 
-      <Route exact path="/" component={Home} />
-      <Route path="/about" component={About} />
-      <Route path="/topics" component={Topics} />
-    </div>
-  </Router>
-);
+```js
+<Consumer>
+ { value => { /* render something based on the context value */ } }
+</Consumer>
 ```
 
-By passing down some information from the `Router` component, each `Link` and `Route` can communicate back to the containing `Router`.
-
-Before you build components with an API similar to this, consider if there are cleaner alternatives. For example, you can pass entire React components as props if you'd like to.
+A React component that subscribes to context changes.
 
-## Referencing Context in Lifecycle Methods
-
-If `contextTypes` is defined within a component, the following [lifecycle methods](/docs/react-component.html#the-component-lifecycle) will receive an additional parameter, the `context` object:
-
-- [`constructor(props, context)`](/docs/react-component.html#constructor)
-- [`componentWillReceiveProps(nextProps, nextContext)`](/docs/react-component.html#componentwillreceiveprops)
-- [`shouldComponentUpdate(nextProps, nextState, nextContext)`](/docs/react-component.html#shouldcomponentupdate)
-- [`componentWillUpdate(nextProps, nextState, nextContext)`](/docs/react-component.html#componentwillupdate)
+Requires a [function as a child](/docs/render-props.html#using-props-other-than-render). This function receives the current context value and returns a React node. It will be called whenever the Provider's value is updated.
```

Comment:
"whenever the Provider's value is updated" is a bit vague. We need to be very explicit that new and old Provider value is compared with `Object.is` algorithm, and whenever it changes, all consumers are re-rendered.